File: iniciante/python/1118-varias-notas-com-validacao.py
def validacao_de_nota():
    while True:
        soma_notas, valida_nota = calculo_das_somas()
        media = soma_notas / valida_nota
        print(f'media = {media:.2f}')
        novo_calculo = calculo_novo()
        if novo_calculo == 2:
            break


def calculo_das_somas():
    soma_das_notas = notas_validas = 0
    while True:
        if notas_validas == 2:
            break
        nota = float(input())
        if 0.0 <= nota <= 10.0:
            soma_das_notas += nota
            notas_validas += 1
        else:
            print('nota invalida')
    return soma_das_notas, notas_validas


def calculo_novo():
    while True:
        continuar = int(input('novo calculo (1-sim 2-nao)\n'))
        if continuar == 1 or continuar == 2:
            break
    return continuar


validacao_de_nota()

File: iniciante/c/1118-varias-notas-com-validacao.c
#include <stdio.h>
#include <stdbool.h>

int main(){
    int i = 0, decisao, count = 0;
    float nota, soma = 0, media;
    while (i < 2){
        scanf("%f", &nota);
        if (nota < 0.0 || nota > 10.0){
            printf("nota invalida\n");
        }
        else{
            soma += nota;
            i++;
            count++;
        }
        
        if (i == 2){
            media = soma / count;
            printf("media = %.2f\n", media);
            while (true){
                printf("novo calculo (1-sim 2-nao)\n");
                scanf("%d", &decisao);
                if (decisao == 1 || decisao == 2){
                    i = 0;
                    break;
                }
            }
        }
    }

    return 0;
}
